allennlp/predictors/coref.py

```diff
@@ -70,6 +72,55 @@ def predict_tokenized(self, tokenized_document: List[str]) -> JsonDict:
         """
         instance = self._words_list_to_instance(tokenized_document)
         return self.predict_instance(instance)
+    
+    @staticmethod
+    def replace_corefs(document: Doc, clusters: List[List[int]]) -> str:
+        """
+        Uses a list of coreference clusters to convert a spacy document into a 
+        string, where each coreference is replaced by its main mention. 
+        """
+    
+        resolved = list(tok.text_with_ws for tok in document)
+        
+        for cluster in clusters:
+            mention_start, mention_end = cluster[0][0], cluster[0][1] + 1
+            mention_span = document[mention_start: mention_end]
+            
+            for coref in cluster[1:]:
+                final_token = document[coref[1]]
+                # Correctly formats possessive mentions with 's endings
+                # These include my, his, her, as well as persons's, computer's, etc.
+                if final_token.tag_ in ["PRP$", "POS"]:
+                    resolved[coref[0]] = mention_span.text + "'s" + final_token.whitespace_
```

Comment:
Perhaps Spacy avoids reconstructing a sentence because the possessive suffix depends on the resolved word.  E.g. "Michael's" (singluar) but "zebras'" (plural).  Your logic seems like a reasonable compromise, but I would expect a clear disclaimer.

Reply:
Yes, that is one area where mistakes in grammar might occur, in producing "zebras's". In the spaCy neuralcoref extension, you're right that it doesn't add this reconstruction, and instead replaces possessive pronouns with their main mention directly, e.g. "the country's" becomes "China". I can see how certain applications would benefit from maintaining the possessive form, while others would do fine without it. More `if..else` style code could be added to guarantee grammatical accuracy in these cases, but it's probably stretching the logic too far and opening up further sources of error.

Reply:
Yep--it would be stretching the logic too far.  I recommend just adding a disclaimer about the issue in the comments of this method.